CoreUtilities/src/au/gov/asd/tac/constellation/utilities/svg/SVGData.java

```diff
@@ -0,0 +1,346 @@
+     * @param idValue 
+     * @return child
+     */
+    public SVGData getChild(final String idValue) {
+        SVGData child = this.children.get(idValue);
+        if (child == null) {
+            for (SVGData childIndex : this.getAllChildren()) {
+                child = childIndex.getChild(idValue);
+                if (child != null) {
+                    return child;
+                }
+            }
+        }
+        return child;
+    }
+    
+    /**
+     * Removes an SVGData with a specified id attribute value.
+     * Use depth first search.
+     * @param idValue 
+     * @return child
+     */
+    public SVGData removeChild(final String idValue) {   
+        //Try and remove the child from this SVGObjects set fo children.
+        SVGData child = this.children.remove(idValue);
+        
+        //If the child wasnt removed, recursively itterate through levels of children untill the id is found.
+        if (child == null) {
+            for (SVGData childIndex : this.getAllChildren()) {
+                child = childIndex.removeChild(idValue);
+
+                //Break out of te search once the ID is found.
+                if (child != null) {
+                    break;
+                }
+            }
+        }
+        
+        // If a child was ound sever the childs reference to the parent.
+        if (child != null){
+            child.setParent(null);
+        }
+        return child;
+    }
+    
+    /**
+     * Returns a List of SVGObjects containing all child SVGData elements.
+     * @return children
+     */
+    public List<SVGData> getAllChildren() {
+        final List<SVGData> allChildren = new ArrayList<>();
+        children.keySet().forEach(key -> allChildren.add(this.children.get(key)));
+        if (!allChildren.isEmpty() && allChildren.get(0).attributes.keySet().contains(SVGAttributeConstant.CUSTOM_SORT_ORDER.getName())) {
+            Collections.sort(allChildren, (SVGData first, SVGData second) -> {
+                String firstValue = first.getAttributeValue(SVGAttributeConstant.CUSTOM_SORT_ORDER.getName());
+                String secondValue = second.getAttributeValue(SVGAttributeConstant.CUSTOM_SORT_ORDER.getName());
+                return Float.compare(Float.parseFloat(secondValue), Float.parseFloat(firstValue));
+            });
+        }
+        return allChildren;
+    }
+    
+    /**
+     * Associates SVGData as a parent of this SVGData.
+     * By extension the parent sets this object as its child. 
+     * Warning: Child object must have a unique ID.
+     * @param parent
+     */
+    public final void setParent(final SVGData parent) {
+        final String idAttributeName = SVGAttributeConstant.ID.getName(); 
+        
+        if (parent != null) {            
+            if (parent.getChild(this.getAttributeValue(idAttributeName)) != null){
+                throw new ArrayIndexOutOfBoundsException(String.format("Parent SVGData %s already has a child of id %s", 
```

Comment:
similar case with Null pointer, not sure if array out of the bounds is the correct exception here. Perhaps `IllegalStateException`?